### ISSUE TITLE
clippy does not fire warning for ok_or usage 

### ISSUE THREAD

**Issue by aliciawyy** [rust-lang/rust-clippy#12517]
### Summary

When running the provided code in the Rust Playground, the expectation was for Clippy to produce a warning related to the usage of `ok_or` in accordance with the `or_fun_call` lint. However, Clippy did not generate any warning, contrary to the expected behavior.

### Reproducer

I tried this code:

```rust
fn foo() -> String {
    println!("Called foo");
    "foo".to_string()
}

fn bar(x: bool) -> Option<i32> {
    if x {
        Some(1)
    } else {
        None
    }
}

fn main() {
  let v = bar(true).ok_or(foo());
  println!("v {:?}", v)
}
```

Playground demo: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=1cc70f21798932cb2cae892524e8cb20

I expected to see this happen:
I expect clippy to fire a warning for the `ok_or` usage as per [or_fun_call](https://rust-lang.github.io/rust-clippy/rust-1.76.0/#/or_fun_call).

Instead, this happened:
Clippy doesn't fire any warning

**Comment by y21** [comment]
The lint is in the nursery category since #9829, so it is allow-by-default and needs to be opted into.

If you put `#![warn(clippy::or_fun_call)]` at the crate root, then you'll see the warning that you are expecting.

**Comment by aliciawyy** [comment]
I see. Thank you!
